experimental/PropertyDDS/packages/property-binder/test/data_binder/data_binding.spec.js

```diff
@@ -2333,3 +2393,98 @@ describe('DataBinding.registerOnPath() should work for', function() {
+
+  beforeAll(function() {
+    registerTestTemplates();
+
+    PropertyFactory.register(Vector3DSchema);
+    PropertyFactory.register(Object3DSchema);
+  });
+
+  beforeEach(async function() {
+    // console.log('inner before each');
+    dataBinder = new DataBinder();
+    workspace = await MockSharedPropertyTree();
+  });
+
+  afterEach(function() {
+    // Unbind checkout view
+    dataBinder.detach();
+
+    // Unregister DataBinding paths
+    _.forEach([
+      ParentDataBinding,
+      ChildDataBinding,
+      PrimitiveChildrenDataBinding,
+      InheritedChildDataBinding,
+      DerivedDataBinding
+    ],
+    unregisterAllOnPathListeners
+    );
+
+    dataBinder = null;
+  });
+
+  it('Map of primitives', function() {
+    var mapInsertSpy = jest.fn();
+    var mapModifySpy = jest.fn();
+    var mapRemoveSpy = jest.fn();
+    ParentDataBinding.registerOnValues('mapPrimitive', ['collectionInsert'], mapInsertSpy);
+    ParentDataBinding.registerOnValues('mapPrimitive', ['collectionModify'], mapModifySpy);
+    ParentDataBinding.registerOnValues('mapPrimitive', ['collectionRemove'], mapRemoveSpy);
+
+    // Register the base (Child) typeid
+    dataBinder.register('BINDING', MapContainerTemplate.typeid, ParentDataBinding);
+    dataBinder.attachTo(workspace);
+
+    var mapContainerPset = PropertyFactory.create(MapContainerTemplate.typeid, 'single');
+    workspace.root.insert('myMapContainerTemplate', mapContainerPset);
+
+    // Expect the insertion of map values to trigger onInsert messages
+    var mapProperty = workspace.root.get(['myMapContainerTemplate', 'mapPrimitive']);
+    workspace.pushNotificationDelayScope();
+    mapProperty.insert('one', '1');
+    mapProperty.insert('two', '2');
+    workspace.popNotificationDelayScope();
+
+    expect(mapInsertSpy).toHaveBeenCalledTimes(2);
+    // Test first parameter (index or key)
+    expect(mapInsertSpy.mock.calls[0][0]).toEqual('one');
+    expect(mapInsertSpy.mock.calls[1][0]).toEqual('two');
+
+    // Test second parameter
+    expect(mapInsertSpy.mock.calls[0][1]).toEqual('1');
+    expect(mapInsertSpy.mock.calls[1][1]).toEqual('2');
+
+    mapInsertSpy.mockClear();
+
+    // modify map
+    workspace.pushNotificationDelayScope();
+    mapProperty.setValues({'one': '10'});
+    mapProperty.setValues({'two': '20'});
+    workspace.popNotificationDelayScope();
+    expect(mapModifySpy).toHaveBeenCalledTimes(2);
+    expect(mapModifySpy.mock.calls[0][0]).toEqual('one');
+    expect(mapModifySpy.mock.calls[1][0]).toEqual('two');
+
+    expect(mapModifySpy.mock.calls[0][1]).toEqual('10');
+    expect(mapModifySpy.mock.calls[1][1]).toEqual('20');
+    mapModifySpy.mockClear();
+
+    // remove from map
+    workspace.pushNotificationDelayScope();
+    mapProperty.remove('one');
+    mapProperty.remove('two');
+    workspace.popNotificationDelayScope();
+    expect(mapRemoveSpy).toHaveBeenCalledTimes(2);
+    expect(mapRemoveSpy.mock.calls[0][0]).toEqual('one');
```

Comment:
Above you wrote that the parent is not available upon removal and thus this should be undefined, but here you are actually expecting the value. Is the comment above incorrect?

Reply:
This is the key/index and not the value